### PR TITLE
Envoy 0.0.1 release

### LINKED ISSUE #8: authentication system

### LINKED ISSUE #9: deckbuilder

### LINKED ISSUE #10: decks dashboard

### LINKED ISSUE #13: setup client-side code coverage with istanbul-middleware

### LINKED ISSUE #14: setup server-side code coverage with istanbul

### LINKED ISSUE #15: setup client-side unit test runner

### LINKED ISSUE #16: setup server-side unit test runner

### LINKED ISSUE #17: setup integration test runner

### LINKED ISSUE #18: setup linter

### PR DESCRIPTION
**Pull request by lambdabaa**
### Overview
- authentication system, close #8 
- deckbuilder, close #9 
- decks dashboard, close #10 
- setup client-side code coverage with istanbul-middleware, close #13 
- setup server-side code coverage with istanbul, close #14 
- setup client-side unit test runner, close #15 
- setup server-side unit test runner, close #16 
- setup integration test runner, close #17 
- setup linter, close #18

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+machine:
```
Comment (lightsofapollo) [inline circle.yml:1]: How do you like circle CI?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "glob": "3.2.x",
+    "grunt": "0.4.x",
+    "grunt-cli": "0.1.x",
+    "grunt-contrib-jshint": "0.8.x",
+    "grunt-mocha": "0.4.x",
+    "grunt-mocha-test": "0.9.x",
+    "istanbul": "0.2.x",
+    "istanbul-middleware": "0.1.x",
+    "meteorite": "0.7.x",
+    "mocha": "1.17.x",
```
Comment (lightsofapollo) [inline package.json:21]: mocha 1.18 has built in promise support which will make you happy

File: test/index.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+      req.send(JSON.stringify(window.__coverage__));
+    });
+  </script>
+
+  <script src="./unit/mocks/ejson.js"></script>
+  <script src="./unit/mocks/session.js"></script>
+  <script src="./unit/mocks/template.js"></script>
+
+  <!-- TODO(gareth): Should be able to run client-side unit tests without
```
Comment (lightsofapollo) [inline test/index.html:38]: FWIW- Karma is pretty good I still am not happy with it (or any other client side testing thing)

File: circle.yml (hunk 1 above)
Comment (lambdabaa) [inline circle.yml:1]: It's awesome. Also the guy who's running it used to work on spidermonkey :)

File: package.json (hunk 2 above)
Comment (lambdabaa) [inline package.json:21]: Oh nice! I can probably get rid of http://selenium.googlecode.com/git/docs/api/javascript/module_selenium-webdriver_testing.html

File: test/index.html (hunk 3 above)
Comment (lambdabaa) [inline test/index.html:38]: Oh cool. It's sort of bittersweet when I find out that I've written a bunch of crappy glue that I maybe didn't need to haha

**Pull request merged**